### PR TITLE
Prohibit bare tuples

### PR DESCRIPTION
**Pull request by catlee**
We've had many production failures due to inadvertently having trailing commas on a line, resulting in a tuple where the single object was expected.

I'd like a way to check for bare trailing commas. This PR reuses the existing C819 error for this case, but I'm open to creating a new error code for it.

### CODE REVIEW

**Comment by graingert** [comment]
I'd probably make it C818 so I can release it as 1.1.0.

Can you make a changelog entry, doc changes, etc?

also I think there's already a flake8 plugin for this though.

**Comment by graingert** [comment]
@catlee ah it's here https://github.com/ar4s/flake8_tuple

**Comment by graingert** [comment]
@ar4s how do you feel about rolling your rule into this checkout?
@catlee can you port all the tests from https://github.com/ar4s/flake8_tuple ?

**Comment by catlee** [comment]
Ah, thanks!
Yeah, I don't mind porting the code/tests. Let me know which way we want to go.

**Comment by graingert** [comment]
@catlee it would be a good idea to port the tests so we can find out any edge cases.

**Comment by catlee** [comment]
Do we want to reuse the same error code/message?

**Comment by graingert** [comment]
@catlee for now whatever is easiest to get the tests running.  Try and port the tests to match the style in this repo, rather than copying them verbatim from @ar4s

**Comment by ar4s** [comment]
@catlee @graingert I'm glad to see my test in this repo :)

**Comment by graingert** [comment]
@catlee if you commit as `chris@atlee.ca` or add `catlee@mozilla.com` as a uid to https://keyserver.ubuntu.com/pks/lookup?op=vindex&search=0x79835F835202491614B8129427EF48554428DB0C&fingerprint=on github will show your commits as verified

File: flake8_commas/_base.py [hunk 1]
```diff
@@ -260,6 +260,7 @@ def get_comma_errors(tokens):
                 (stack[-1].comma not in TUPLE_ISH or stack[-1].n > 1)
             ) or stack[-1].comma == LAMBDA_EXPR and token.type == COLON
         )
+
```
Comment (graingert) [inline flake8_commas/_base.py]: whitespace only change

**Comment by graingert** [comment]
It's probably worth changing the title from:

`Flake8 Extension to enforce trailing commas.`
to

`Flake8 Extension to enforce comma placement.`

or something similar

**Comment by graingert** [comment]
@sigmavirus24 any thoughts on this?

**Comment by catlee** [comment]
"Prohibit trailing commas on bare tuples" is probably more accurate.

```1, 2``` is fine, but ```1, 2,``` is not.

**Comment by graingert** [comment]
Yep that's a better error message. I'm just thinking what should be the name for the whole repo?

**Comment by sigmavirus24** [comment]
I, personally, *never* liked bare tuple syntax except in explicit assignment, e.g.,

```py
a, b = b, a
```

And that seems unaffected by this rule. I know bare tuples have caused folks a **lot** of pain, so I'm :+1: on this as a matter of "helping others help themselves". For folks that don't like this, it can always be added to their ignore list.

**Comment by graingert** [comment]
I've released this as 2.0.0rc1
